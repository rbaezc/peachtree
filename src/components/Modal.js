import React from 'react';

const Modal = ({ handleClose, show, children }) => {
    const showHideClassName = show ? "modal d-block" : "modal d-none";

    return (
        <div className={showHideClassName}>
          <div className="modal-container">
            {children}
            <a href="!#" className="modal-close btn" onClick={handleClose}>
              x
            </a>
          </div>
        </div>
    );
}

export default Modal;