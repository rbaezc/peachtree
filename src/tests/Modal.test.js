import React from 'react';
import renderer from 'react-test-renderer';
import Modal from '../components/Modal';

test('Modal hides when click on close button', () => {
    const component = renderer.create(
        <Modal></Modal>
    );

    let modal = component.toJSON();
    expect(modal).toMatchSnapshot();

    // manually trigger handleClose
    modal.props.handleClose();
    // re-rendering
    modal = component.toJSON();
    expect(modal).toMatchSnapshot();
})


test('Modal shows when show option is true' , () => {
    const component = renderer.create(
        <Modal show={true}></Modal>
    );

    let modal = component.toJSON();
    expect(modal).toMatchSnapshot();

    // manually trigger show
    modal.props.show();
    // re-rendering
    modal = component.toJSON();
    expect(modal).toMatchSnapshot();
})