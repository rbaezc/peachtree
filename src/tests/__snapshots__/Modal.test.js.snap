// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal hides when click on close button 1`] = `
<div
  className="modal d-none"
>
  <div
    className="modal-container"
  >
    <button
      className="modal-close"
      type="button"
    >
      close
    </button>
  </div>
</div>
`;

exports[`Modal shows when show option is true 1`] = `
<div
  className="modal d-block"
>
  <div
    className="modal-container"
  >
    <button
      className="modal-close"
      type="button"
    >
      close
    </button>
  </div>
</div>
`;
